### ISSUE TITLE
How to get player email?

### ISSUE THREAD

**Issue by CrazyFlasher** [myflashlab/GameServices-ANE#14]
How to get email?
For ex. I've added GET_ACCOUNTS permission but how to get it?

**Comment by myflashlab** [comment]
Well, first you should add [the permission]. Then you need to ask users to grant your app the permission like [this](https://github.com/myflashlab/GameServices-ANE/blob/master/FD/src/LoginSample.as#L180). Then you should listen to ```GameServices.google.auth.addEventListener(AuthEvents.AUTH, onAuthDataRetrived);``` and finally try loginin. you'll have the token and the email address of the user like [this](https://github.com/myflashlab/GameServices-ANE/blob/master/FD/src/LoginSample.as#L379-L384)

```actionscript
private function onAuthDataRetrived(e:AuthEvents):void
{
	trace("name = " + e.auth.name);
	trace("email = " + e.auth.email);
	trace("scopes = " + e.auth.scopes);
	trace("token = " + e.auth.token);
}
```

**Comment by CrazyFlasher** [comment]
Did like you suggested, but e.auth is null for some reasons

**Comment by CrazyFlasher** [comment]
Did like you suggested, but e.auth is still null:
1. added permission to manifest
2. asked to grant permission before GameServices.init
3. popup appeared and I picked "allow"

1.
```
private function checkForContacts():void
{
	var permissionState:int = _exPermissions.check(PermissionCheck.SOURCE_CONTACTS);
		if (permissionState == PermissionCheck.PERMISSION_UNKNOWN || permissionState == PermissionCheck.PERMISSION_DENIED)
	{
		_exPermissions.request(PermissionCheck.SOURCE_CONTACTS, onContactsRequestResult);
	}
	else
	{
		continueInit();
	}
}

private function onContactsRequestResult($state:int):void
{
	if ($state != PermissionCheck.PERMISSION_GRANTED)
	{
		trace("You did not allow the app the required permissions!");
	}
	else
	{
		continueInit();
	}
}
```
2.
```
private function continueInit():void
{
	// initialize the Game Services and wait for a successful init call before doing anything else
	GameServices.init(false); // set this to 'true' when you are building for production
	GameServices.google.auth.addEventListener(AuthEvents.INIT, onInit);
}

private function onInit(e:AuthEvents):void
{
	var result:String;
		switch (e.status) // status code indicating whether Game Services can run in your app or not. It can be one of following results:
	{
		case GameServices.SUCCESS:
				result = "GameServices.SUCCESS";
			        onSuccessfullInit();
				break;
...		
private function onSuccessfullInit():void
{
	_initialized = true;
		// add required listeners for the authentication process of the Game Services ANE
	GameServices.google.auth.addEventListener(AuthEvents.TRYING_SILENT_LOGIN, onSilentTry);
	GameServices.google.auth.addEventListener(AuthEvents.LOGIN, onLoginSuccess);
	GameServices.google.auth.addEventListener(AuthEvents.LOGOUT, onLogout);
	GameServices.google.auth.addEventListener(AuthEvents.ERROR, onLoginError);
	GameServices.google.auth.addEventListener(AuthEvents.CANCELED, onCanceled);
	GameServices.google.auth.addEventListener(AuthEvents.SETTING_WINDOW_DISMISSED, onSettingWinDismissed);
		// Do you stuff here after you are sure that Game Services is supported and can be used in your app.
}		
```
3.
```
private function onLoginSuccess(e:AuthEvents):void
{
	trace("onLoginSuccess");
		// some information about current logged in user.
	trace("displayName = " + e.player.displayName);
	trace("id = " + e.player.id);
	if (e.auth)
	{
		_email = e.auth.email;
		trace("email = " + e.auth.email);
			}

			_userId = e.player.id;

			_signedIn = true;

			dispatchMessage(SignInModelMessage.SIGN_IN_SUCCESS, null, true);
		}
```

**Comment by CrazyFlasher** [comment]
also trying to get e.auth in this callback:
```
private function onInit(e:AuthEvents):void
{
	var result:String;
		if (e.auth)
	{
		_email = e.auth.email;
		trace("email = " + e.auth.email);
	}
...
```
but it is still null there

**Comment by myflashlab** [comment]
Ok, it seems like Google has removed the ```GoogleAuthUtil.getToken``` method because some reasons! it's good to know about this. We are working on a fix but in the meanwhile please test this quick built ANE and see if you can get the email address or not? https://drive.google.com/file/d/0B7eHG2CEml2TQ2EtdzR1dlllUTg/view?usp=sharing

This ANE will not return the token anymore but at least the user email is returned like before. Let me know how it goes while we are trying to find a way to fix the token thing. 

Thanks for reporting this recent bug.

**Comment by CrazyFlasher** [comment]
Hi, I replaced gamesServices.ane with quickDemo_gameServices.ane and same result.
auth object in event is still null

**Comment by myflashlab** [comment]
After a successful login, call ```GameServices.google.auth.getAuth();``` while listening to ```GameServices.google.auth.addEventListener(AuthEvents.AUTH, 						onAuthDataRetrived);```

then you will receive the email:

```actionscript
private function onAuthDataRetrived(e:AuthEvents):void
{
	trace("name = " + e.auth.name);
	trace("email = " + e.auth.email);
	trace("scopes = " + e.auth.scopes);
	trace("token = " + e.auth.token);
}
```

**Comment by CrazyFlasher** [comment]
thanks! that worked

**Comment by myflashlab** [comment]
Released V2.2.1 with this problem fixed.

**Comment by MyMyGames** [comment]
Why there is no response at all after login when calling  authorization's  function  "GameServices.google.auth.getAuth();" ?

`private function onLoginSuccess(e:AuthEvents):void
{
	trace("onLoginSuccess");
	GameServices.google.auth.addEventListener(AuthEvents.AUTH, onAuthDataRetrived);
	GameServices.google.auth.getAuth();
}`
^^^^^^^
it's ok!

`private function onAuthDataRetrived(e:AuthEvents):void
{
	trace("name = " + e.auth.name);
	trace("email = " + e.auth.email);
	trace("scopes = " + e.auth.scopes);
	trace("token = " + e.auth.token);
}`
^^^^^^^
no reaction (
